JSTutorial/hw_7/task_2.js

```diff
@@ -0,0 +1,299 @@
+    })
+}
+getCommonInfo(enterprises);
+
+//2
+function getFactoryByNameORId(key) {
+    let name = null;
+    enterprises.forEach(enterprise => {
+        enterprise.departments.forEach(department => {
+            if (department.id === key || department.name === key) name = enterprise.name;
+        })
+    })
+    return name;
+}
+console.log(getFactoryByNameORId("Отдел тестирования"))
+console.log(getFactoryByNameORId(6))
+
+//3
+function addEnterprise(name) {
+    const nejObj = {
+        name,
+        departments: [],
+        id : enterprises.length - 1,
+    }
+    enterprises.push(nejObj);
+}
+
+addEnterprise("Предприятие 4")
+console.log(enterprises);
+
+//4
+function getEnterpriseByIdOrName(key) {
+    let enterpriseObj = null;
+    enterprises.forEach(enterprise => {
+        if (enterprise.id === key || enterprise.name === key) enterpriseObj = enterprise
+    })
+    return enterpriseObj;
+}
+function addDepartment(id, name) {
+    const enterprise = getEnterpriseByIdOrName(id);
+    const nejObj = {
+        name,
+        employees_count: [],
+        id : enterprise.id + 12,
+    }
+    enterprise.departments.push(nejObj);
+}
+addDepartment(2, "Новый отдел");
+console.log(enterprises[3])
+
+//5
+function editDepartment(id, name) {
+    const enterprise = getEnterpriseByIdOrName(id);
+    enterprise.name = name;
+}
+editDepartment(2, "Измененное предприятие")
+console.log(enterprises[3])
+
+//6
+function deleteEnterprise(id) {
+    const index = enterprises.findIndex(enterprise => enterprise.id === id);
+    if (index !== -1) {
+        enterprises.splice(index, 1);
+    }
+}
+
+deleteEnterprise(2)
+console.log(enterprises)
+
+//7
+function getDepartmentIndexById(id) {
+    let department = enterprises.map(enterprise => enterprise.departments.find(department => department.id === id))
+    return department.filter(value => value !== undefined);
+}
+function deleteDepartment(id) {
+    let index;
+    enterprises.forEach(enterprise => {
+        index = enterprise.departments.findIndex(department => department.id === id && department.employees_count !== 0)
+        if (index !== -1) {
+            enterprise.departments.splice(index, 1);
+        }
+    });
+}
+delete deleteDepartment(4);
+console.log(enterprises[0])
+
+//9
+function findEmployeeByDepartmentIdAndSetNewValue(id, setValue) {
+    let count;
+    enterprises.forEach(enterprise => {
+        enterprise.departments.map(department => {
+            if (department.id === id) {
+                count = department.employees_count;
+                if (setValue !== "undefined") {
+                    if (setValue === 0) {
+                        department.employees_count = setValue;
+                    } else department.employees_count += setValue;
+                }
+            }
+        })
+    });
+    return count;
+}
+
+function moveEmployees(idFrom, idTo) {
```

Comment:
As an alternative because you've already made all necessary functions
```js
    const from = getDepartment(whereFromDepartID, enterprises);
    const to = getDepartment(whereToDepartID, enterprises);
    if (from && to && getEnterpriseByDeptIDorName(whereFromDepartID) === getEnterpriseByDeptIDorName(whereToDepartID)) {
        to.employees_count += from.employees_count;
        from.employees_count = 0;
    } else { /* */ }
```